### PR TITLE
Debuginfo lights

### PR DESCRIPTION
**Pull request by dkollmann**
The debug view (Alt + D) now shows all the lights in the scene.

### CODE REVIEW

**Comment by elishacloud** [comment]
Is there a reason why the debug items are in the release build?  Can we move them over to the debug build?

Also, I like to keep these items inside the `Dd7to9` clause since they cannot be used in DirectDraw natively,

**Comment by dkollmann** [comment]
The thing is, that I need them in the release build. The proper solution would probably be to have a shipping or development configuration.

Ah yeah makes sense. What do you think about copying the data to the device object instead of calling DirectX functions? I wanted to avoid running functions which need to enter locks.

**Comment by elishacloud** [comment]
> What do you think about copying the data to the device object instead of calling DirectX functions? 

Yes, great idea.  We can also put them into another class and file to keep the main file cleaner.

**Comment by elishacloud** [comment]
I'll merge these changes and then move them into a new class afterwards.

File: ddraw/IDirect3DDeviceX.cpp [hunk 1]
```diff
@@ -1807,7 +1861,35 @@ HRESULT m_IDirect3DDeviceX::SetLight(DWORD dwLightIndex, LPD3DLIGHT7 lpLight)
 			}
 		}
 
-		return (*d3d9Device)->SetLight(dwLightIndex, &Light);
+		hr = (*d3d9Device)->SetLight(dwLightIndex, &Light);
```
Comment (PatrickvL) [inline ddraw/IDirect3DDeviceX.cpp:1864]: `hr` could have been declared here instead of above

File: ddraw/IDirect3DDeviceX.cpp [hunk 2]
```diff
@@ -1848,7 +1931,21 @@ HRESULT m_IDirect3DDeviceX::LightEnable(DWORD dwLightIndex, BOOL bEnable)
 			return DDERR_GENERIC;
 		}
 
-		return (*d3d9Device)->LightEnable(dwLightIndex, bEnable);
+		hr = (*d3d9Device)->LightEnable(dwLightIndex, bEnable);
```
Comment (PatrickvL) [inline ddraw/IDirect3DDeviceX.cpp:1934]: `hr` could have been declared here instead of above

Comment (elishacloud) [reply ddraw/IDirect3DDeviceX.cpp:1934]: Good call.  Will fix this in the next update.

**Comment by elishacloud** [comment]
I moved the debug code to a new class.  See here:  552c9837a1ded3f3faabed8a6a35cdf10b554a86

**Pull request merged**